src/instance.rs

```diff
@@ -890,12 +890,19 @@ impl<'py, T> BoundObject<'py, T> for Bound<'py, T> {
     }
 }
 
-/// A borrowed equivalent to `Bound`.
+/// A borrowed equivalent to [`Bound`].
 ///
-/// The advantage of this over `&Bound` is that it avoids the need to have a pointer-to-pointer, as Bound
-/// is already a pointer to an `ffi::PyObject``.
+/// [`Borrowed<'a, 'py, T>`] is an advanced type used just occasionally at the edge of interaction
+/// with the Python interpreter. It can be thought of as analogous to the shared reference `&'a
+/// Bound<'py, T>`, similarly this type is `Copy` and `Clone`. The difference is that [`Borrowed<'a,
+/// 'py, T>`] is just a smart pointer rather than a reference-to-a-smart-pointer. For one this
+/// reduces one level of pointer indirection, but additionally it removes the implicit lifetime
+/// relation that `'py` has to outlive `'a` (`'py: 'a`). This opens the possibility to borrow from the
+/// underlying Python object without necessarily requiring attachment to the interpreter for that
+/// duration. Within PyO3 this is used for example for the byte slice (`&[u8]`) extraction.
 ///
-/// Similarly, this type is `Copy` and `Clone`, like a shared reference (`&T`).
```

Comment:
Might also be worth mentioning that some Python C APIs return "borrowed" pointers which this type models (though those APIs are falling out of favour with free-threading, maybe except on tuples).